### PR TITLE
AboveThresholdProcessor

### PR DESCRIPTION
**Pull request by relleums**
As we wished for. With basic unit tests.

### CODE REVIEW

File: src/main/java/fact/extraction/AboveThreshold.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    protected double threshold = 1800;
+
+    @Override
+    public Data process(Data input) {
+
+        final int npix = (Integer) input.get("NPIX");
+        final int roi = (Integer) input.get("NROI");
+        final short[] timeSerieses = (short[]) input.get(dataKey);
+        final short thresholdShort = (short)threshold;
+        ArrayList<Integer> pixelsAboveThresholdList = new ArrayList<>();
```
Comment (MaxNoe) [inline src/main/java/fact/extraction/AboveThreshold.java]: You should use a pixelset here, not a plain int list

**Comment by MaxNoe** [comment]
I changed the Processor to use a PixelSet and also save slice information, please have a look.

File: src/main/java/fact/extraction/AboveThreshold.java [hunk 2]
```diff
@@ -37,25 +42,29 @@ public Data process(Data input) {
 
         final int npix = (Integer) input.get("NPIX");
         final int roi = (Integer) input.get("NROI");
-        final short[] timeSerieses = (short[]) input.get(dataKey);
+        final short[] timeSeries = (short[]) input.get(dataKey);
```
Comment (relleums) [inline src/main/java/fact/extraction/AboveThreshold.java]: I can not wrap my head around this. There is no sense in a single time series for all the independent pixels. Why not having a plural 's'?

File: src/test/java/fact/extraction/AboveThresholdTest.java [hunk 3]
```diff
@@ -39,10 +51,15 @@ public void testAboveThresholdCertainPixelsAbove() {
...
+        PixelSet testPixels = new PixelSet();
+        testPixels.addById(42);
+        testPixels.addById(314);
+        testPixels.addById(1337);
+
         for(int pix=0; pix<1440; pix++) {
             for(int slice=0; slice<300; slice++) {
                 final int pos = pix*300 + slice;
-                if(pix==42 || pix==314 || pix==1337) {
+                if((pix==42 || pix==314 || pix==1337) && (slice == 271 || slice == 256)) {
```
Comment (relleums) [inline src/test/java/fact/extraction/AboveThresholdTest.java]: Looks good and simple to also test the slices.

**Comment by relleums** [comment]
It looks like you managed to combine our demands. Since you told me that the PixelSet knows how to convert itself into a simple list when appended to the output json writer, I approve all this.

**Comment by MaxNoe** [comment]
Series is both plural and singular. There is no word serieses


Am 23. Januar 2017 8:18:32 nachm. schrieb "Sebastian A. Mueller" 
<notifications@github.com>:

File: src/main/java/fact/extraction/AboveThreshold.java (hunk 2 above)
Comment (MaxNoe) [inline src/main/java/fact/extraction/AboveThreshold.java]: Series is both the singular and plural form. There is now serieses.

**Comment by MaxNoe** [comment]
It's actually not the `PixelSet` itself but the `JSONWriter` who knows how to handle a `PixelSet` through a `TypeAdapter`;)

File: src/main/java/fact/extraction/AboveThreshold.java (hunk 2 above)
Comment (relleums) [inline src/main/java/fact/extraction/AboveThreshold.java]: I propose to replace all timeseries(es) with timelines  or sth. else which has a sane plural representation. We should not limit ourselves by using variable names which have no flexibility or restricted features.

File: src/main/java/fact/extraction/AboveThreshold.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                      "The fraction of slices above the threshold is put into outputKey + SliceRatio" +
+                      "The number of slices above the threshold is put into outputKey + SliceCount"
+    )
+    private String outputKey = null;
+
+    @Parameter(
+        required = true,
+        description = "Threshold value to be exceeded"
+    )
+    protected double threshold = 1800;
```
Comment (jebuss) [inline src/main/java/fact/extraction/AboveThreshold.java]: set this to null

**Pull request merged**